fulltext/__init__.py

```diff
@@ -98,6 +125,9 @@ def _get_file(backend, f, **kwargs):
     backend's `_get_file()` if one is provided. Otherwise, it will write the
     data to a temporary file and call `_get_path()`.
     """
+    if not is_binary(f):
+        raise AssertionError('File must be opened in binary mode.')
```

Comment:
I would use `ValueError` just in case somebody uses `-O` switch

Reply:
`-O` disables `assert`, not `AssertionError`.

```
$ python -O
>>> assert 1 == 2, 'Foobar'
>>> raise AssertionError()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AssertionError
```